### DISCUSSION TITLE
Allow location android permission prompt not coming in ionic3 while using  navigator.geolocation.getCurrentPosition()

### DISCUSSION THREAD

**Discussion by nitinkrs** [community/community#64162] [Programming Help]
### Select Topic Area

General

### Body

I am using **ionic**-3 ,**cordova**-8 & **cordova-android**-11 and using the  "**cordova-plugin-geolocation**": "^4.0.1" in my code but didnot able to get the location prompt coming from device side while opening the app ,

I have also used **Android permissions** in my **config.xml** & **Android Manifest.xml** but didn't get then also.
 <plugin name="cordova-plugin-geolocation" spec="4.0.1">`
      <variable name="GEOLOCATION_USAGE_DESCRIPTION" value="App uses geolocation to determine where you are." />
</plugin>
    <platform name="android">
        <uses-permission name="android.permission.ACCESS_COARSE_LOCATION" />
        <uses-permission name="android.permission.ACCESS_FINE_LOCATION" />
        <uses-permission name="android.permission.ACCESS_LOCATION_EXTRA_COMMANDS" />
    </platform>

**Comment by zainmujahid** [comment]
If you're not seeing the location permission prompt in your Ionic 3 app while using `navigator.geolocation.getCurrentPosition()`, follow these steps to troubleshoot:

1. **Plugin Installation:** Confirm that `cordova-plugin-geolocation` is properly installed using `ionic cordova plugin ls`.

2. **Permissions in `config.xml`:** Make sure you've added `<uses-permission>` for `ACCESS_COARSE_LOCATION` and `ACCESS_FINE_LOCATION`.

3. **Permissions in `AndroidManifest.xml`:** Verify that the same permissions are present in the `AndroidManifest.xml`.

4. **Runtime Permission Request:** Before using `getCurrentPosition()`, request permissions at runtime using `cordova.plugins.permissions`.

5. **Device Location Settings:** Ensure device location services are enabled and set to "High accuracy" or "Device only" mode.

6. **Test on Real Device:** Emulators might behave differently, so test on an actual device.

7. **Update Plugin:** Use the latest version of `cordova-plugin-geolocation` and other relevant plugins.

**Reply by nitinkrs** [reply]
I have tried all above steps which you mentioned but didnot able to get the location access prompt coming from device side to allow location
